### PR TITLE
Ensureloggedin

### PR DESCRIPTION
**Pull request by ba5eem**
Updated auth in App, newitemform, server, login, logout, reducer login. Added new routes for login, logout, register - this is a before lunch PR to be continued with more specificity after lunch on UX and correct auth pages views based of client req

### CODE REVIEW

**Comment by ba5eem** [comment]
[Add login, logout, register api route](https://trello.com/c/GB1ubHmz/72-add-login-logout-register-api-route)

**Comment by ba5eem** [comment]
[include redirect on index.js](https://trello.com/c/BrKbo1YB/70-include-redirect-on-indexjs)

**Comment by ba5eem** [comment]
[react route auth vs unauth](https://trello.com/c/JoabbAyC/68-react-route-auth-vs-unauth)

File: server/routes/login.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      
+      local.id = req.user.id;
+      local.username = req.user.username;
+      local.auth = true;
+      return res.json(local);
+    });
+  })(req, res, next);
+});
+
+function isAuthenticated(req, res, next){
```
Comment (taesup) [inline server/routes/login.js]: This doesn't do anything?

**Review by taesup — CHANGES_REQUESTED** [review]
You should remove all console logs that are no longer needed.

**Comment by taesup** [comment]
Looks good to me! Merge It!
ship it

**Pull request merged**